cli/src/lib/createStub.js

```diff
@@ -0,0 +1,178 @@
+      if (err) {
+        reject(err);
+      } else {
+        resolve(result);
+      }
+    })
+  );
+}
+
+const stubTemplate =
+`declare module '%s' {
+  declare module.exports: any;
+}`;
+
+function stubFor(moduleName: string): string {
+  return format(stubTemplate, moduleName);
+}
+
+async function writeStub(
+  projectRoot: string,
+  packageName: string,
+  version: string,
+  overwrite: boolean,
+  files: Array<string>,
+): Promise<string> {
+  let output = stubFor(packageName);
+  if (files.length > 0) {
+    output += format(
+      "\n\n%s\n",
+      files
+        .map(file => file.replace(/\.[^.]+$/, '')) // trim extensions
+        .map(file => stubFor(format("%s/%s", packageName, file)))
+        .join("\n\n"),
+    );
+  }
+  output += "\n"; // File should end with a newline
+  const filename = path.join(projectRoot, "flow-typed", packageName + ".js");
+  await mkdirp(path.dirname(filename));
+
+  if (!overwrite) {
+    const exists = await fs.exists(filename);
+    if (exists) {
+      const existingStub = await fs.readFile(filename);
+      if (!verifySignedCode(existingStub.toString())) {
+        throw new Error(
+          "Stub already exists and has been modified. " +
+          "Use --overwrite to overwrite",
+        );
+      }
+    }
+  }
+
+  await fs.writeFile(filename, signCode(output, version));
+  return filename;
+}
+
+/**
+ * createStub("/path/to/root", "foo") will create a file
+ * /path/to/root/flow-typed/foo.js that contains a stub for the module foo.
+ *
+ * If foo is installed, it will read the directory that require("foo") resolves
+ * to and include definitions for "foo/FILE", for every FILE in the foo package
+ */
+export default async function createStub(
+  projectRoot: string,
+  packageName: string,
+  explicitVersion: string | null,
+  overwrite: boolean,
+): Promise<boolean> {
+  let files = [];
+  let resolutionError = null;
+  let pathToPackage = null;
+  let version = explicitVersion || null;
+  try {
+    pathToPackage = await resolve(packageName, {basedir: process.cwd()});
+
+    let dir = path.dirname(pathToPackage);
+    while (path.basename(dir) !== "node_modules") {
+      pathToPackage = dir;
+      dir = path.dirname(pathToPackage);
+
+      if (dir === pathToPackage) {
+        throw new Error("Failed to find package directory");
+      }
+    }
+
+    files = await glob("**/*.{js,jsx}", {
+      cwd: pathToPackage,
+      ignore: "node_modules/**",
+    });
+
+  } catch (e) {
+    resolutionError = e;
+  }
+
+  // Try to deduce a version if one isn't provided
+  if (version == null) {
+    // Look at the package.json for the installed module
+    if (pathToPackage != null) {
+      try {
+        version = (require: any)(path.join(pathToPackage, 'package.json')).version;
+      } catch (e) {}
+    }
+
+    // If that failed, try looking for a package.json in the root
+    if (version == null) {
+      (require: any)(path.join(projectRoot, 'package.json')).version;
```

Comment:
Err...this looks like dead copy-pasta. Reading the version and throwing it away doesn't do very much :P